python2/typing.py

```diff
@@ -1081,6 +1065,38 @@ def __init__(self, *args, **kwargs):
         super(GenericMeta, self).__init__(*args, **kwargs)
         if isinstance(self.__extra__, abc.ABCMeta):
             self._abc_registry = self.__extra__._abc_registry
+            self._abc_cache = self.__extra__._abc_cache
+        elif self.__origin__ is not None:
+            self._abc_registry = self.__origin__._abc_registry
+            self._abc_cache = self.__origin__._abc_cache
```

Comment:
I could not just remove them, since some other places expect their presence. However, I found a bug in ``_abc_negative_cache`` setters, and I made most assignments to ``_abc_negative_cache`` a no-op. Could you please check the memory/speed situation once more with the newest commit in my branch?